### PR TITLE
fix: add native cues when using native text tracks

### PR DESCRIPTION
**Pull request by third774**
Co-authored-by: Kyle Boutette <kyleveb@gmail.com>

## Description
 

When native text tracks are used, the cues added need to be native cues.

Related to this PR in the player: https://github.com/videojs/video.js/pull/6410

## Specific Changes proposed
Checks to see if native tracks are being used, and if so converts the cue into a native cue.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [x] Example created ([starter template on JSBin](https://jsbin.com/gejugat/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

**Comment by third774** [comment]
Not sure how best to go about creating an example with this fix implemented. It's also dependent on the related PR in the player.

**Comment by third774** [comment]
Additionally, not sure what adding tests for this would look like. With a little guidance I'd be happy to take a stab at it! 😄

**Comment by third774** [comment]
Here's a demo of native captions working with this change!

The `video.js` file in this demo was built with `video.js` on commit `ff9fd1c518396e59c2a0164df1d600d7edb1928f` which is the related PR https://github.com/videojs/video.js/pull/6410 and having linked the `@videojs/http-streaming` dep to this PR's commit `86dfba96a3cb3f812caae2bcf25bd40365572221`

https://third774.github.io/videojs-native-captions-fix/

**Comment by gkatsev** [comment]
Thanks @third774, we'll take a look soon.

**Comment by third774** [comment]
Looks like this breaks a few tests... looking into this now.

File: src/vtt-segment-loader.js [hunk 1]
```diff
@@ -361,7 +361,9 @@ export default class VTTSegmentLoader extends SegmentLoader {
     segmentInfo.cues.forEach((cue) => {
       // remove any overlapping cues to prevent doubling
       this.remove(cue.startTime, cue.endTime);
-      this.subtitlesTrack_.addCue(cue);
+      this.subtitlesTrack_.addCue(this.hls_.player_.tech_.featuresNativeTextTracks ?
```
Comment (third774) [inline src/vtt-segment-loader.js]: So, it looks like the mock config that is used in the tests does not have the `hls_` property.

We could either
1. Add this to the mock
2. Check the full path to the `featuresNativeTextTracks` property (see below)

```js
this.hls_
  && this.hls_.player_
  && this.hls_.player_.tech_
  && this.hls_.player_.tech_.featuresNativeTextTracks
```
@gkatsev do you have a preference for which? Should we also add some tests for when this is `true` vs `false`?

Comment (gkatsev) [reply src/vtt-segment-loader.js]: After some discussion, a better way will likely to pass a `featuresNativeTextTracks` flag when we create the `VTTSegmentLoader` in the `master-playlist-controller` https://github.com/videojs/http-streaming/blob/450ca91ade3c8fd7b96834e811ccd6699daee834/src/master-playlist-controller.js#L172-L175
Adding something like:
```js
featuresNativeTextTracks: this.tech_.featuresNativeTextTracks
```
as an option next to `loaderType`. Then setting is a local option in the VTTSegmentLoader constructor https://github.com/videojs/http-streaming/blob/86dfba96a3cb3f812caae2bcf25bd40365572221/src/vtt-segment-loader.js#L36

**Comment by third774** [comment]
Added a test for this as well!

**Comment by third774** [comment]
Just wanted to circle back around on this — is there anything else blocking this change (and related PR https://github.com/videojs/video.js/pull/6410) from being merged?

**Comment by gkatsev** [comment]
@third774 hey, thanks for your patience! We'll take a look in the next day or so.

**Comment by gkatsev** [comment]
Closing this in favor of https://github.com/videojs/http-streaming/pull/759, I rebased it and has a tiny change in the test. I'll merge it manually to maintain authorship credit once it's reviewed.

**Pull request closed without merging**